contracts/vaults/TreasuryVault/RibbonTreasuryVaultWithAirSwap.sol

```diff
@@ -0,0 +1,1165 @@
+            VaultLifecycleTreasuryWithAirSwap.CloseParams({
+                OTOKEN_FACTORY: OTOKEN_FACTORY,
+                USDC: USDC,
+                currentOption: oldOption,
+                delay: DELAY,
+                lastStrikeOverrideRound: lastStrikeOverrideRound,
+                overriddenStrikePrice: overriddenStrikePrice,
+                period: period
+            });
+
+        (address otokenAddress, uint256 strikePrice, uint256 delta) =
+            VaultLifecycleTreasuryWithAirSwap.commitAndClose(
+                strikeSelection,
+                closeParams,
+                vaultParams,
+                vaultState
+            );
+
+        emit NewOptionStrikeSelected(strikePrice, delta);
+
+        optionState.nextOption = otokenAddress;
+
+        uint256 nextOptionReady = block.timestamp.add(DELAY);
+        require(
+            nextOptionReady <= type(uint32).max,
+            "Overflow nextOptionReady"
+        );
+        optionState.nextOptionReadyAt = uint32(nextOptionReady);
+
+        _closeShort(oldOption);
+
+        // In case chargeAndDistribute was not called last round, call
+        // the function to conclude last round's performance fee and distribution
+        if (IERC20(USDC).balanceOf(address(this)) > MIN_DUST_AMOUNT) {
+            _chargeAndDistribute();
+        }
+    }
+
+    /**
+     * @notice Closes the existing short position for the vault.
+     */
+    function _closeShort(address oldOption) private {
+        uint256 lockedAmount = vaultState.lockedAmount;
+        if (oldOption != address(0)) {
+            vaultState.lastLockedAmount = uint104(lockedAmount);
+        }
+        vaultState.lockedAmount = 0;
+
+        optionState.currentOption = address(0);
+
+        if (oldOption != address(0)) {
+            uint256 withdrawAmount =
+                VaultLifecycleTreasuryWithAirSwap.settleShort(GAMMA_CONTROLLER);
+            emit CloseShort(oldOption, withdrawAmount, msg.sender);
+        }
+    }
+
+    /**
+     * @notice Rolls the vault's funds into a new short position.
+     */
+    function rollToNextOption() external onlyKeeper nonReentrant {
+        (
+            address newOption,
+            uint256 lockedBalance,
+            uint256 queuedWithdrawAmount
+        ) = _rollToNextOption(uint256(lastQueuedWithdrawAmount));
+
+        lastQueuedWithdrawAmount = queuedWithdrawAmount;
+
+        ShareMath.assertUint104(lockedBalance);
+        vaultState.lockedAmount = uint104(lockedBalance);
+
+        emit OpenShort(newOption, lockedBalance, msg.sender);
+
+        uint256 mintAmount =
+            VaultLifecycleTreasuryWithAirSwap.createShort(
+                GAMMA_CONTROLLER,
+                MARGIN_POOL,
+                newOption,
+                lockedBalance
+            );
+
+        IERC20 optionToken = IERC20(newOption);
+        optionToken.safeApprove(address(AIRSWAP_CONTRACT), mintAmount);
```

Comment:
Since the variable optionToken is only used once - suggestion would be to simplify to 1 line:

`IERC20(newOption).safeApprove(address(AIRSWAP_CONTRACT), mintAmount);`

Reply:
Fixed in [a30c9d0](https://github.com/ribbon-finance/ribbon-v2/pull/297/commits/a30c9d05c95b9d301e7c16500455108d96450e59)